### PR TITLE
Add 3 files to export

### PR DESCRIPTION
**Pull request by TheRealPaco**
I am writting a packge for [eosio signing request (esr) ](https://github.com/eosio-eps/EEPs/blob/master/EEPS/eep-7.md#ESR---The--EOSIO-Signing-Request--protocol) and I need to import the following files (serialize.dart, eosdart_base.dart, transaction.dart) to create the signing request model and serialize and deserialize the request. Would you mind adding those 3 exports to the package. it would be better for me then using a fork repository like I do for now.

Thanks

### CODE REVIEW

File: lib/src/client.dart [hunk 1]
```diff
@@ -351,6 +351,21 @@ class EOSClient {
...
     return PushTransactionArgs(signatures, serializedTrx);
   }
+
+  Future<AccountNames> getKeyAccountsState(String pubKey) async {
+    http.Response response = await http.post(
+        '${this._nodeURL}/${this._version}/state/get_key_accounts',
+        body: {"public_key": pubKey});
+
+    if (response.statusCode >= 300) {
+      throw response.body;
```
Comment (fzxu) [inline lib/src/client.dart]: can we do better instead of throw the body?

**Review by fzxu — CHANGES_REQUESTED** [review]

**Comment by fzxu** [comment]
any reason to ref to the fork?

**Comment by TheRealPaco** [comment]
> any reason to ref to the fork?

Sorry for not responding earlier, 
I am working on a port for greymass anchor-link and I need to recover an eos public key from some signed data, so I added the feature to eosdart_ ecc and dart told me me that the eosdart package and my dart-anchor-link package need to reference the same eosdart_ecc. So for testing, I just changed it to use my fork for simplicity's sake and it updated the pull request which I had kind of forgot about. I will create a pull request for eosDart_ecc and I will change it back. After that I will change it to use the official eosDart_ecc package and also update eosDart with your other comment so we can look into merging it.

File: lib/src/client.dart (hunk 1 above)
Comment (TheRealPaco) [reply lib/src/client.dart]: Finnaly, I don't need this anymore so I'll remove it from the pull request

**Comment by fzxu** [comment]
@TheRealPaco I've merged that change and release 0.4.3

**Comment by 7flash** [comment]
Hello @TheRealPaco :)
I am working on dartesr package that has similar purpose, do you want to collaborate?

https://github.com/digital-scarcity/dartesr

File: lib/src/serialize.dart [hunk 2]
```diff
@@ -598,9 +598,9 @@ void serializeStruct(Type self, SerialBuffer buffer, Object data,
     self.base.serialize(self.base, buffer, data,
         state: state, allowExtensions: allowExtensions);
   }
-  var dy = data as dynamic;
+  Map dy = data as dynamic;
   for (var field in self.fields) {
-    if (dy[field.name] != null) {
+    if (dy.containsKey(field.name)) {
```
Comment (TheRealPaco) [inline lib/src/serialize.dart:603]: In eosjs serialize `if (field.name in data)` check if the map contain the field not if the field is not null. Deserializing a null value work fine but this was causing an error when serializing null value.

**Comment by TheRealPaco** [comment]
> Hello @TheRealPaco :)
> I am working on dartesr package that has similar purpose, do you want to collaborate?
> 
> https://github.com/digital-scarcity/dartesr

Hi, 
Yes we can but this is my last week full time on this as I was working as an intern for EOS Nation and I am going back to university. I might do some hours here and there during my session depending on my availability.

For [dart-esr](https://github.com/EOS-Nation/dart-esr), the master branch work fine but I have redone it to be more lke greymass eosio signing request js version (for easier integration with the anchor link port to dart). The package is working and almost complete for a v1. Right now, it's more finalizing things, cleaning the code, adding tests, and add documentation. There is example though if you want to take a look at the up top date version [https://github.com/EOS-Nation/dart-esr/tree/feature/complete-port-of-esr-js](https://github.com/EOS-Nation/dart-esr/tree/feature/complete-port-of-esr-js). 

Also, I'll probably be merging the branch soon and pushing it to pub.dev.

**Comment by fzxu** [comment]
@TheRealPaco let me know once the PR is ready for review.

**Comment by TheRealPaco** [comment]
> @TheRealPaco let me know once the PR is ready for review.

It is ready. finally I do not need many changes, just exposing those 4 files and fixing the serializing bug with null value. If I have more things in the future,I'll do another PR.

**Review by fzxu — APPROVED** [review]

**Pull request merged**